### ISSUE TITLE
Small model occupies too much GPU in CUDA11.1 + Torch1.8.1 but is normal in Torch 1.6 + CUDA10.1

### ISSUE THREAD

**Issue by bo-miao** [pytorch/pytorch#55277]
## 🐛 Bug

At the forward step, pass a small ResNet model occupies me more than 15G GPU with torch1.8. Torch1.6 does not have this issue using the same code.

## To Reproduce

1. I reimplement the code https://github.com/zlai0/MAST on DAVIS dataset and their pretrained model.
2. I first tested the code on the first environment with Torch1.6 + CUDA10.1 + 2080Ti, with the --ref 2 as said in the README.md of the repo, the model only requires about **6G GPU** memory.
3. Then I tested the code on 3090, Torch1.8 + CUDA11.1, with the --ref 2, the model requires about **20G GPU** memory. There is a large memory gap in the two environments.
4. I tracked the memory changes and found that the memory cost (more than 15G) was located at the backbone (ResNet18).
5. Memory significantly changes code place (https://github.com/zlai0/MAST -> .\models\mast.py -> 32 row: feats_r = [self.post_convolution(self.feature_extraction(rgb)) for rgb in rgb_r]. ( This step contains a ResNet18 with a small convolution layer)
6. I test the code 32 row in CPU environment using torchstat stat(self.feature_extraction, (3, 480, 910)) and found that ResNet only occupies 1241.54MB in CPU.

Thanks!

## Expected behavior
How to solve the issue that GPU memory cost significantly increase when passing the small model and what happened there. My 3090 does not support the low version PyTorch, so I have to solve this issue.

## Environment

PyTorch version: 1.8.1+cu111
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 11.2.142
GPU models and configuration: GPU 0: GeForce RTX 3090
Nvidia driver version: 460.67
cuDNN version: Probably one of the following:
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.8.1+cu111
[pip3] torchstat==0.0.7
[pip3] torchvision==0.9.1+cu111
[conda] blas                      1.0                         mkl    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
[conda] mkl                       2017.0.3                      0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
[conda] numpy                     1.19.5                   pypi_0    pypi
[conda] torch                     1.8.1+cu111              pypi_0    pypi
[conda] torchstat                 0.0.7                    pypi_0    pypi
[conda] torchvision               0.9.1+cu111              pypi_0    pypi

```

 - PyTorch Version (e.g., 1.0): 1.8.1
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6.2
 - CUDA/cuDNN version: 11.1
 - GPU models and configuration: code at https://github.com/zlai0/MAST, inference on DAVIS dataset with their pretrained model.
 - Any other relevant information:

Thanks!

**Comment by mch5048** [comment]
I am facing the same issue. Given the same shaped tensor inputs, running with RTX 3090 and 1.8.1 binary + cu111 occupies the GPU memory around 3~4 times (compared to model ran on Titan RTX with torch 1.6.0 + cu101).

**Comment by bo-miao** [comment]
Even I do not know what exactly happened, but the GPU Memory Cost is reduced from ~20G to ~6G on 3090 after adding **torch.cuda.empty_cache()** to clean the memory and using **torch.cuda.amp.autocast** on the ResNet18 backbone.

**Comment by Sanster** [comment]
Some possibly relevant links

- https://github.com/microsoft/onnxruntime/issues/9289
- https://forums.developer.nvidia.com/t/higher-memory-usage-ampere-compared-to-pascal/198966

**Comment by bo-miao** [comment]
> @bo-miao Did you solve this problem?
> 
> I also did some benchmarks(resnet18 from torchvision), the result is that on the 3090, the model's GPU memory usage will be 70% more than 1080
> 
> <img alt="image" width="715" src="https://user-images.githubusercontent.com/3998421/162870290-eb9792d6-3f07-4c56-a5bd-c3be63a0c1ae.png">
> 
> Here is the benchmark script, requirements: torch, torchvision, py3nvml
> 
> ```python
> import time
> import torch
> from py3nvml import nvidia_smi
> import torchvision.models as models
> 
> print(torch.__version__)
> 
> allow_tf32 = True
> print(f"allow_tf32: {allow_tf32}")
> torch.backends.cuda.matmul.allow_tf32 = allow_tf32
> torch.backends.cudnn.allow_tf32 = allow_tf32
> 
> 
> nvidia_smi.nvmlInit()
> device_id = 0
> handle = nvidia_smi.nvmlDeviceGetHandleByIndex(device_id)
> 
> 
> model = models.resnet18(pretrained=False).cuda()
> model = model.eval()
> 
> with torch.no_grad():
>     dummy_input = torch.randn(10, 3, 640, 640, device="cuda")
> 
>     # warmup
>     for _ in range(10):
>         model(dummy_input)
> 
>     start = time.time()
>     for _ in range(100):
>         model(dummy_input)
>     print(f"time: {(time.time() - start)*1000}")
> 
>     memory_mb = nvidia_smi.nvmlDeviceGetMemoryInfo(handle).used / 1024 / 1024
> 
>     print(f"memory_mb: {memory_mb}")
>     nvidia_smi.nvmlShutdown()
> ```

The problem still has not been solved...

**Comment by Sanster** [comment]
I found that on different GPU, the memory occupied by initializing CUDA Session has **huge** difference

```python
import torch
a = torch.Tensor(1).cuda()
```

|torch|torchvision|GPU|GPU Memory(mb)|DIff(%)|
|-----|-----------|-----|------------------|-------|
|1.8.2+cu101|0.9.2+cu101|1080|721| 0 |
|1.8.2+cu111|0.9.2+cu111|3090|1687| 133% |
